docs/source/explanations/caching.md

```diff
@@ -0,0 +1,100 @@
+# Caching Design and Roadmap
+
+For practical guides on client-side and service-side caching, see
+{doc}`../tutorials/caching` and {doc}`../how-to/tune-caches`.
+
+```{note}
+This page discusses both current features and planned features. Italicized remarks
+in the discussion below makes clear what exists now and what is on the road map.
+```
+
+## Overview
+
+There are three types of caches in Tiled:
+
+1. **Client-side response cache.** The Tiled Python client implements a standard
+   web cache, similar in both concept and implementation to a web browser's cache.
+   It enables an offline "airplane mode". If a server is available, it enables the
+   client to check whether the version it has is the latest one.
+2. **Service-side response cache.**
+   _This is not yet implemented, but planned soon._  This is also a standard web
+   cache, on the server side. It stores the content of the most frequent responses.
+   This covers use cases such as, "Several users are asking for the exact same
+   chunks of data in the exact same format."
+3. **Service-side data cache.** The _response_ caches are very finely targeted.
+   Requests that ask for overlapping but distinct slices of data, or requests
+   that ask for the same but in varied formats, will not benefit from that
+   cache; they will "miss". The _data_ cache, however, stores chunks of (array,
+   dataframe) data, which is it can slice and encode differently for different
+   requests. It will not provide quite the same speed boost as a response cache,
+   but it has a broader impact.
```

Comment:
Maybe something here about how 3 stores in-memory python objects? Also what would you think about calling it `Server-side object cache`?

Reply:
Also...how about something about how it can be used by those coding readers?

Reply:
I'll grant you that "data cache" is a bit broad. I already went through and renamed it from "internal cache" (because of course it may not always be internal). I'm 50/50 on "object" being better, and I don't really have the heart to rename it again now but I am open to renaming it in the future.

Reply:
> Maybe something here about how 3 stores in-memory python objects?

I mention it below, but I think you are right it should be included here as well. Adding...

Reply:
I could foresee whee we tighten this down a bit, actually, and don't allow just _any_ Python object to be dumped in here. Our current usage is just numpy arrays and dataframes, which give us more options for serializing to Redis that we'll have if we fully embrace "anything Python".

Reply:
For the record: we agreed to sit on it and consider other names. "Data" is definitely overly broad --- all of these caches store data.